lib/money/helpers.rb

```diff
@@ -46,16 +46,20 @@ def value_to_currency(currency)
         currency
       when nil, ''
         default = Money.current_currency || Money.default_currency
-        raise(ArgumentError, 'missing currency') if default.nil? || default == ''
+        raise(Money::Currency::UnknownCurrency, 'missing currency') if default.nil? || default == ''
```

Comment:
This specific exception is handled by `legacy_default_currency` which will set the default currency to a NullCurrency (if none is set) and not raise here. 
I'm ok with changing this to instead have a deprecation here when `legacy_deprecation` is enabled, otherwise raise.
Not sure which is easier to manage, I'm good either way